# Cosmetics & Beauty Products - Sales Analysis

## Objective
Analyze global cosmetics and beauty product data to extract actionable business insights for the beauty retail industry.

## Business Questions
1. Is there a relationship between price and rating? (Are expensive products better rated?)
2. Which categories (skincare, makeup, etc.) have the best price/quality ratio?
3. How does price vary by skin type or gender target?
4. What percentage of products are cruelty-free, and does it vary by brand or country?
5. Which countries of origin have the most presence and how do they compare in rating/price?

## Dataset
Source: Kaggle - Top Beauty & Cosmetics Products Worldwide 2024

## df.info()
It shows you a general summary of the dataset: rows, columns, data type.

In [2]:
import pandas as pd
df = pd.read_csv('../data/raw/most_used_beauty_cosmetics_products.csv')
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 15000 entries, 0 to 14999
Data columns (total 14 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Product_Name       15000 non-null  object 
 1   Brand              15000 non-null  object 
 2   Category           15000 non-null  object 
 3   Usage_Frequency    15000 non-null  object 
 4   Price_USD          15000 non-null  float64
 5   Rating             15000 non-null  float64
 6   Number_of_Reviews  15000 non-null  int64  
 7   Product_Size       15000 non-null  object 
 8   Skin_Type          15000 non-null  object 
 9   Gender_Target      15000 non-null  object 
 10  Packaging_Type     15000 non-null  object 
 11  Main_Ingredient    15000 non-null  object 
 12  Cruelty_Free       15000 non-null  bool   
 13  Country_of_Origin  15000 non-null  object 
dtypes: bool(1), float64(2), int64(1), object(10)
memory usage: 1.5+ MB


## Initial Exploration
- 15,000 products with no missing values
- Price and Rating are numeric, ready for analysis
- Cuelty_Free is boolean, easy to filter

## df.describe()
It shows you basic statistics for the numeric columns: minimun, maximun, mean, median.

In [3]:
df.describe()

,Price_USD,Rating,Number_of_Reviews
count,15000.000000,15000.000000,15000.000000
mean,80.134108,3.002327,5014.231333
std,40.402983,1.168029,2855.665464
min,10.000000,1.000000,52.000000
25%,45.480000,2.000000,2562.000000
50%,80.040000,3.000000,5002.000000
75%,114.760000,4.000000,7497.000000
max,149.990000,5.000000,10000.000000


## Statistical Summary
- Average price: $80 USD (range: $10 - $150)
- Ratings evenly distributed from 1 to 5 (avg: 3.0)
- Reviews range from 52 to 10,000 per product

## df['Brand'].unique()
Counts the number of unique brands in the dataset.
## df['Category'].unique()
Displays all the different categories, useful for spotting inconsistencies.
## df['Country_of_Origin'].unique()'
Same principle, checks for entries like "USA" and "United States", which represent the same thing but are written differently.

In [7]:
print(df['Brand'].unique())
print(df['Category'].unique())
print(df['Country_of_Origin'].unique())
print(df['Usage_Frequency'].unique())
print(df['Skin_Type'].unique())
print(df['Gender_Target'].unique())
print(df['Packaging_Type'].unique())
print(df['Main_Ingredient'].unique())
print(df['Cruelty_Free'].unique())


['Drunk Elephant' 'Laura Mercier' 'Natasha Denona' 'Ilia Beauty'
 'Charlotte Tilbury' 'Danessa Myricks' 'Bourjois' 'IT Cosmetics'
 'Fenty Beauty' 'Sisley' 'Juvia’s Place' 'NARS' 'ColourPop' 'Huda Beauty'
 'Tatcha' 'Kiehl’s' 'Tarte' 'Glossier' 'Make Up For Ever'
 'Anastasia Beverly Hills' 'E.l.f.' 'Hourglass' 'Pat McGrath Labs'
 'Too Faced' 'Perricone MD' 'RMS Beauty' 'Urban Decay' 'Rare Beauty'
 'Becca' 'Patrick Ta' 'Shiseido' 'Kylie Cosmetics' 'Bite Beauty'
 'Yves Saint Laurent' 'Bobby Brown' 'Farsali' 'Morphe' 'Milk Makeup'
 'Clinique' 'KVD Beauty']
['Blush' 'Makeup Remover' 'Highlighter' 'Face Mask' 'Foundation' 'Powder'
 'Lip Gloss' 'CC Cream' 'Eye Shadow' 'Concealer' 'Eyeliner' 'Lipstick'
 'Setting Spray' 'Cleanser' 'Bronzer' 'Primer' 'Face Oil' 'Contour'
 'Mascara' 'Serum' 'BB Cream' 'Exfoliator' 'Lip Liner' 'Moisturizer']
['Australia' 'UK' 'Italy' 'South Korea' 'Germany' 'Japan' 'USA' 'France']
['Weekly' 'Occasional' 'Daily' 'Monthly']
['Sensitive' 'Dry' 'Normal' 'Oily' 'Combina

 ## Dataset Overview
 - 40 unique brands (mix of luxury and mass market)
 - 24 product categories (makeup and skincare)
 - 8 countries of origin
 No null values, no duplicates, dataset is clean and ready for analysis.

## df.duplicated().sum()
Counts how many rows are completely duplicated, identical rows that should be removed.

In [8]:
df.duplicated().sum()

np.int64(0)

## Export Clean Data

In [9]:
import os
os.makedirs('../data/processed', exist_ok=True)
df.to_csv('../data/processed/cosmetics_clean.csv', index=False)


## Load Data into SQLite
SQLite is a database that resides in a single file on your computer, in this case, 'cosmetics.db'.

In [10]:
import sqlite3
conn = sqlite3.connect('../data/cosmetics.db')
df.to_sql('products', conn, if_exists='replace', index=False)
conn.close()

## Verifying that the data imported correctly: Why do we verifiy that the import was succesful?
Because imports can sometimes fail silently, the data comes in but is poorly formatted, has misnamed columns, or is incomplete.

In [11]:
conn = sqlite3.connect('../data/cosmetics.db')
result = pd.read_sql('SELECT * FROM products LIMIT 5', conn)
conn.close()
result

,Product_Name,Brand,Category,Usage_Frequency,Price_USD,Rating,Number_of_Reviews,Product_Size,Skin_Type,Gender_Target,Packaging_Type,Main_Ingredient,Cruelty_Free,Country_of_Origin
0,Ultra Face Mask,Drunk Elephant,Blush,Weekly,67.85,1.4,686,30ml,Sensitive,Female,Tube,Retinol,0,Australia
1,Ultra Lipstick,Laura Mercier,Makeup Remover,Occasional,116.43,4.2,5483,250ml,Dry,Unisex,Bottle,Shea Butter,0,UK
2,Ultra Serum,Natasha Denona,Highlighter,Daily,90.84,1.6,5039,100ml,Sensitive,Male,Compact,Aloe Vera,1,Italy
3,Divine Serum,Ilia Beauty,Face Mask,Occasional,55.17,3.2,6202,250ml,Normal,Male,Tube,Glycerin,1,South Korea
4,Super Foundation,Charlotte Tilbury,Highlighter,Occasional,140.56,1.7,297,100ml,Oily,Female,Compact,Glycerin,0,Germany


## SQL Analysis - Business Questions
### Question 1: Is there a relationship between price and rating? (Are expensive products better rated?)

In [26]:
conn = sqlite3.connect('../data/cosmetics.db')
result = pd.read_sql('''
		SELECT 
			CASE
				WHEN Price_USD < 50 THEN '1-Low ($10-$50)'
				WHEN Price_USD < 100 THEN '2-Mid ($50-$100)'
				ELSE '3-High ($100-$150)'
			END AS Price_Range,
			ROUND(AVG(Rating), 2) AS Avg_Rating,
			COUNT(*) AS Num_Products
		FROM products
		GROUP BY Price_Range
		ORDER BY Price_Range ASC
''', conn)
conn.close()
result

,Price_Range,Avg_Rating,Num_Products
0,1-Low ($10-$50),3.03,4297
1,2-Mid ($50-$100),2.97,5305
2,3-High ($100-$150),3.01,5398


### Results: Price vs Rating
No significant relationship found between price and rating.
ALL price ranges average 3.0 stars, suggesting that higher price does not guarantee better costumer satisfaction.

### Question 2: Which categories (skincare, makeup, etc.) have the best price/quality ratio?

In [27]:
conn = sqlite3.connect('../data/cosmetics.db')
result = pd.read_sql('''
		SELECT Category,
		ROUND(AVG(Rating), 2) AS Avg_Rating,
		ROUND(AVG(Price_USD), 2) AS Avg_Price,
		ROUND(AVG(Rating)/AVG(Price_USD), 4) AS Value_Index
		FROM products
		GROUP BY Category
		ORDER BY Value_Index DESC
''', conn)
conn.close()
result

,Category,Avg_Rating,Avg_Price,Value_Index
0,Contour,3.13,79.16,0.0395
1,BB Cream,3.06,78.91,0.0388
2,CC Cream,3.07,79.43,0.0387
3,Eyeliner,2.91,75.72,0.0385
4,Serum,2.97,77.41,0.0383
5,Setting Spray,3.05,79.98,0.0381
6,Powder,2.99,78.66,0.0381
7,Cleanser,3.01,79.00,0.0381
8,Concealer,3.00,78.79,0.0380
9,Primer,3.04,80.11,0.0379


### Results: Best Price:Quality Ratio by Category
Contour, BB Cream and CC Cream show the best value index.
Lip products (Lipstick, Lip Liner) and Eye Shadow show the lowest price/quality ratio.
Overall differences are small, suggesting pricing is uniform across categories in this market.

### Question 3: How does price vary by skin type or gender target?

In [28]:
conn = sqlite3.connect('../data/cosmetics.db')
result = pd.read_sql('''
		SELECT Skin_Type,
		ROUND(AVG(Price_USD), 2) AS Avg_Price
		FROM products
		GROUP BY SKin_Type
		ORDER BY Avg_price DESC
''', conn)
conn.close()
result

,Skin_Type,Avg_Price
0,Dry,81.51
1,Normal,79.98
2,Oily,79.88
3,Combination,79.83
4,Sensitive,79.50


In [29]:
conn = sqlite3.connect('../data/cosmetics.db')
result = pd.read_sql('''
		SELECT Gender_Target,
		ROUND(AVG(Price_USD), 2) AS Avg_Price
		FROM products
		GROUP BY Gender_Target
		ORDER BY Avg_price DESC
''', conn)
conn.close()
result

,Gender_Target,Avg_Price
0,Male,80.26
1,Unisex,80.16
2,Female,79.99


### Results: Price by Skin Type and Gender
- Dry skin products are slightly more expensive ($81.51 avg).
- Male-targeted products marginally more expensive than female ($80.26 vs $79.99).
- Differences are minimal (<$2), suggesting uniform pricing strategy regardless of skin type or gender target.

### Question 4:  What percentage of products are cruelty-free, and does it vary by brand or country?

In [30]:
conn = sqlite3.connect('../data/cosmetics.db')
result = pd.read_sql('''
		SELECT Country_of_Origin,
		ROUND(SUM(Cruelty_Free) * 100.0 / COUNT(*), 2) AS Avg_Cruelty_Free
		FROM products
		GROUP BY Country_of_Origin
		ORDER BY Avg_Cruelty_Free DESC
''', conn)
conn.close()
result

,Country_of_Origin,Avg_Cruelty_Free
0,France,51.58
1,Italy,49.90
2,UK,49.46
3,Japan,49.33
4,Germany,49.15
5,USA,48.83
6,Australia,48.64
7,South Korea,48.19


In [31]:
conn = sqlite3.connect('../data/cosmetics.db')
result = pd.read_sql('''
		SELECT Brand,
		ROUND(SUM(Cruelty_Free) * 100.0 / COUNT(*), 2) AS Avg_Cruelty_Free
		FROM products
		GROUP BY Brand
		ORDER BY Avg_Cruelty_Free DESC
''', conn)
conn.close()
result

,Brand,Avg_Cruelty_Free
0,ColourPop,56.10
1,Glossier,53.35
2,Becca,53.35
3,Farsali,53.13
4,Yves Saint Laurent,53.05
5,Bobby Brown,52.63
6,RMS Beauty,52.62
7,Kylie Cosmetics,52.43
8,Drunk Elephant,51.90
9,Kiehl’s,51.09


### Results: Cruelty-Free by Country and Brand

**By Country:**
France leads with 51.58% cruelty-free products, South Korea lowest at 48.19%.
Minimal variation across countries (~3% difference).

**By Brand**
ColourPop leads with 56.1% cruelty-free products.
Make Up For Ever and Tatcha show lowest cruelty-free rates (~43%).
Overall, all brands hover around 50%, no strong cruelty-free leaders emerge.

### Question 5: Which countries of origin have the most presence and how do they compare in rating/price?

In [32]:
conn = sqlite3.connect('../data/cosmetics.db')
result = pd.read_sql('''
		SELECT Country_of_Origin,
		ROUND(AVG(Rating), 2) AS Avg_Rating,
		ROUND(AVG(Price_USD), 2) AS Avg_Price,
		COUNT(*) AS Presence
		FROM products
		GROUP BY Country_of_Origin
		ORDER BY Presence DESC
''', conn)
conn.close()
result

,Country_of_Origin,Avg_Rating,Avg_Price,Presence
0,Italy,3.00,79.93,1942
1,USA,3.00,80.50,1931
2,Australia,2.96,80.74,1873
3,France,3.01,79.24,1863
4,Japan,3.04,78.77,1861
5,South Korea,3.01,82.40,1851
6,UK,3.01,79.16,1846
7,Germany,3.00,80.34,1833


### Results: Countries by Presence, Rating and Price
- Italy and USA have the highest product count (~1,942 and 1,931).
- Japan has the highest average rating (3.04) and lowest price ($78.77).
- South Korea has the highest average price ($82.40).
- Overall distribution is very uniform across all countries.

## General Conclusion

The analysis reveals uniformly similar results across all business questions:
- Price does not correlate with rating (~3.0 average across all price ranges).
- Category, skin type, gender, and country show minimal price/rating variation.
- Cruelty-Free rates hover around 50% for all brands and countries.

These uniform results are consistent with q synthetically generated dataset where ratings apper randomly distributed (perfect mean of 3.0).
Real market data would show stronger patterns and outliers.

**The value of this project lies in demonstrating a complete data analysis pipeline: Python(pandas) -> SQL (SQLite) -> Tableau (Visualization), and applying a structured analytical methodology to answer business questions.**

## Export the data to Tablaeu

In [34]:
import sqlite3
import pandas as pd
import os

os.makedirs('../data/tableau', exist_ok=True)
conn = sqlite3.connect('../data/cosmetics.db')

# Q1 - Price vs Rating
q1 = pd.read_sql('''
    SELECT 
        CASE
            WHEN Price_USD < 50 THEN '1-Low ($10-$50)'
            WHEN Price_USD < 100 THEN '2-Mid ($50-$100)'
            ELSE '3-High ($100-$150)'
        END AS Price_Range,
        ROUND(AVG(Rating), 2) AS Avg_Rating,
        COUNT(*) AS Num_Products
    FROM products
    GROUP BY Price_Range
    ORDER BY Price_Range DESC
''', conn)
q1.to_csv('../data/tableau/q1_price_vs_rating.csv', index=False)

# Q2 - Category Value Index
q2 = pd.read_sql('''
    SELECT Category,
        ROUND(AVG(Rating), 2) AS Avg_Rating,
        ROUND(AVG(Price_USD), 2) AS Avg_Price,
        ROUND(AVG(Rating)/AVG(Price_USD), 4) AS Value_Index
    FROM products
    GROUP BY Category
    ORDER BY Value_Index DESC
''', conn)
q2.to_csv('../data/tableau/q2_category_value.csv', index=False)

# Q3 - Price by Skin Type and Gender
q3_skin = pd.read_sql('''
    SELECT Skin_Type,
        ROUND(AVG(Price_USD), 2) AS Avg_Price
    FROM products
    GROUP BY Skin_Type
    ORDER BY Avg_Price DESC
''', conn)
q3_skin.to_csv('../data/tableau/q3_skin_type.csv', index=False)

q3_gender = pd.read_sql('''
    SELECT Gender_Target,
        ROUND(AVG(Price_USD), 2) AS Avg_Price
    FROM products
    GROUP BY Gender_Target
    ORDER BY Avg_Price DESC
''', conn)
q3_gender.to_csv('../data/tableau/q3_gender.csv', index=False)

# Q4 - Cruelty Free
q4_country = pd.read_sql('''
    SELECT Country_of_Origin,
        ROUND(SUM(Cruelty_Free) * 100.0 / COUNT(*), 2) AS Cruelty_Free_Pct
    FROM products
    GROUP BY Country_of_Origin
    ORDER BY Cruelty_Free_Pct DESC
''', conn)
q4_country.to_csv('../data/tableau/q4_cruelty_country.csv', index=False)

q4_brand = pd.read_sql('''
    SELECT Brand,
        ROUND(SUM(Cruelty_Free) * 100.0 / COUNT(*), 2) AS Cruelty_Free_Pct
    FROM products
    GROUP BY Brand
    ORDER BY Cruelty_Free_Pct DESC
''', conn)
q4_brand.to_csv('../data/tableau/q4_cruelty_brand.csv', index=False)

# Q5 - Countries Analysis
q5 = pd.read_sql('''
    SELECT Country_of_Origin,
        ROUND(AVG(Rating), 2) AS Avg_Rating,
        ROUND(AVG(Price_USD), 2) AS Avg_Price,
        COUNT(*) AS Num_Products
    FROM products
    GROUP BY Country_of_Origin
''', conn)
q5.to_csv('../data/tableau/q5_countries.csv', index=False)

conn.close()
print("All CSVs exported!")

All CSVs exported!
